core/src/globals.cpp

```diff
@@ -95,4 +98,14 @@ namespace irods::http::globals
 	{
 		return *g_oidc_config;
 	} // oidc_configuration
+
+	auto set_user_mapping_lib(boost::dll::shared_library _lib) -> void
```

Comment:
I think I asked this before, but ... is copying the input arg the expected/correct way of handling the shared lib?

Reply:
Copying the argument is fine, since the original object is created initially in a function, it would go out of scope if we were to catch a reference and not move.

As for why not move immediately, it's mostly for convenience to not have to call the 'getter' for the library.

If you mean is the library okay with multiple instances of a shared library pointing to the same thing, it is. It's how the 'function' style imports work: https://www.boost.org/doc/libs/1_81_0/doc/html/boost_dll/getting_started.html